content/docs/observability/otel-stack.md

```diff
@@ -0,0 +1,680 @@
+   EOF
+   ```
+
+2. Create a Kubernetes ReferenceGrant so that the HTTPListenerPolicy can apply to the OTel logs collector service.
+
+   ```yaml
+   kubectl apply -f- <<EOF
+   apiVersion: gateway.networking.k8s.io/v1beta1
+   kind: ReferenceGrant
+   metadata:
+     name: allow-otel-collector-logs-access
+     namespace: telemetry
+   spec:
+     from:
+     - group: gateway.kgateway.dev
+       kind: HTTPListenerPolicy
+       namespace: {{< reuse "docs/snippets/namespace.md" >}}
+     to:
+     - group: ""
+       kind: Service
+       name: opentelemetry-collector-logs
+   EOF
+   ```
+
+3. Create another HTTPListenerPolicy to collect and store traces in Tempo.
+
+   ```yaml
+   kubectl apply -f- <<EOF
+   apiVersion: gateway.kgateway.dev/v1alpha1
+   kind: HTTPListenerPolicy
+   metadata:
+     name: tracing-policy
+     namespace: {{< reuse "docs/snippets/namespace.md" >}}
+   spec:
+     targetRefs:
+     - group: gateway.networking.k8s.io
+       kind: Gateway
+       name: http
+     tracing:
+       provider:
+         openTelemetry:
+           serviceName: http
+           grpcService:
+             backendRef:
+               name: opentelemetry-collector-traces
+               namespace: telemetry
+               port: 4317
+       spawnUpstreamSpan: true
+   EOF
+   ```
+
+4. Create a Kubernetes ReferenceGrant so that the HTTPListenerPolicy can apply to the OTel traces collector service.
+
+   ```yaml
+   kubectl apply -f- <<EOF
+   apiVersion: gateway.networking.k8s.io/v1beta1
+   kind: ReferenceGrant
+   metadata:
+     name: allow-otel-collector-traces-access
+     namespace: telemetry
+   spec:
+     from:
+     - group: gateway.kgateway.dev
+       kind: HTTPListenerPolicy
+       namespace: {{< reuse "docs/snippets/namespace.md" >}}
+     to:
+     - group: ""
+       kind: Service
+       name: opentelemetry-collector-traces
+   EOF
+   ```
+
+## Step 5: Verify your setup {#verify}
+
+To verify that your setup is working, generate sample traffic and review the logs and Grafana dashboard.
+   
+1. Generate traffic for the httpbin app. 
+
+   {{< tabs items="Cloud Provider LoadBalancer,Port-forward for local testing" tabTotal="2">}}
+   {{% tab tabName="Cloud Provider LoadBalancer" %}}
+   ```sh
+   for i in {1..5}; do curl -v http://$INGRESS_GW_ADDRESS:8080/status/418 -H "host: www.example.com:8080"; done
+   ```
+   {{% /tab %}}
+   {{% tab tabName="Port-forward for local testing" %}} %}} %}}
+   ```sh
+   for i in {1..5}; do curl -v localhost:8080/status/418 -H "host: www.example.com:8080"; done
+   ```
+   {{% /tab %}}
+   {{< /tabs >}}
+
+   Example output:
+
+   ```
+   < HTTP/1.1 418 Unknown
+   I'm a teapot!
+   ...
+   ```
+
+2. Check that logs are being collected.
+
+   ```sh
+   kubectl -n telemetry logs deploy/opentelemetry-collector-logs | grep '/status/418' | wc -l
+   ```
+
+   Example output: The count matches the number of requests that you sent, such as `5`.
+
+   ```
+
```

Comment:
assuming these blank lines are in the ex output intentionally, ignore if so

Reply:
yeah, it is how it rendered in mine, but maybe not needed? it was just weird, so thought I'd include:

```
arthurberger@artberger-MacBook-Pro ~ % kubectl -n telemetry logs deploy/opentelemetry-collector-logs | grep '/status/418' | wc -l

       5
arthurberger@artberger-MacBook-Pro ~ % kubectl -n telemetry logs deploy/opentelemetry-collector-traces | grep 'http.status_code: Str(418)' | wc -l

      10
```